lib/src/command/get.dart

```diff
@@ -52,12 +56,17 @@ class GetCommand extends PubCommand {
       log.warning(log.yellow(
           'The --packages-dir flag is no longer used and does nothing.'));
     }
+
+    if (argResults['dry-run'] && argResults['enforce-lockfile']) {
+      fail('Cannot do a dry-run with enforce-lockfile.');
+    }
```

Comment:
`--dry-run` with `--enforce-lockfile` is fine. In this case it should do a resolution and check that nothing in the lock file changed.
We won't need to download archives, but we could opt to check that content-hashes match what is one the server.